src/WebApiContrib.Core.Versioning/Strategies/IVersionStrategy.cs

```diff
@@ -15,6 +15,6 @@ public interface IVersionStrategy
         /// <param name="context">The <see cref="HttpContext"/> of the current request.</param>
         /// <param name="routeData">The <see cref="RouteData"/>of the current request.</param>
         /// <returns>A version integer, or <c>null</c> if a version cannot be determined.</returns>
-        int? GetVersion(HttpContext context, RouteData routeData);
+        VersionContext GetVersion(HttpContext context, RouteData routeData);
```

Comment:
hey I agree with renaming `VersionContext` => `VersionResult` since it is - after all - a "result". However, not sure about changing it to struct because the version result might or might not be there (depending on whether a request carries version information). 

If it's not there then returning `null` would be applicable, unless the version = 1 is by convention considered a default one and returned instead of null.